### PR TITLE
Feature/scrum 3793

### PR DESCRIPTION
**Pull request by tlm**

### CODE REVIEW

**Comment by apnicjim** [comment]
tests?

File: src/main/java/net/apnic/whowas/intervaltree/Interval.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
     default int compareTo(Interval<K> other) {
         int lc = low().compareTo(other.low());
         if (lc == 0) {
-            lc = high().compareTo(other.high());
+            lc = -1 * high().compareTo(other.high());
```
Comment (apnicjim) [inline src/main/java/net/apnic/whowas/intervaltree/Interval.java]: Good candidate for property based tests

File: src/main/java/net/apnic/whowas/intervaltree/avl/AvlTree.java [hunk 2]
```diff
@@ -38,6 +38,12 @@ private AvlTree(AvlNode<K,V,I> root) {
     }
 
     @Override
+    public Stream<Tuple<I, V>> equalToAndLeastSpecific(I range)
```
Comment (apnicjim) [inline src/main/java/net/apnic/whowas/intervaltree/avl/AvlTree.java:41]: Good candidate for property based tests

Comment (tlm) [reply src/main/java/net/apnic/whowas/intervaltree/avl/AvlTree.java:41]: Done & Done

**Comment by GKTheOne** [comment]
{suggestion} Is it too late to rename the tuple fields to 'first' and 'second' rather than their shorthand 'fst' & 'snd'? Thought they were deep/binary third-party library fields for a bit.

**Review by khloke — APPROVED** [review]

**Review by GKTheOne — APPROVED** [review]

**Pull request merged**